### PR TITLE
pAI Bot Piloting

### PR DESCRIPTION
**Pull request by Geevies**
* pAI can now pilot farmbots, cleanbots, floorbots and medbots.
* Floating messages now appear even if you're inside an object or another mob.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by Chada-1** [comment]
OOF I love this and I've been wanting it for awhile, I tried to port it from /TG/ even and failed miserably.

File: code/modules/mob/living/bot/cleanbot.dm [hunk 1]
```diff
@@ -57,37 +58,36 @@ var/list/cleanbot_types // Going to use this to generate a list of types once th
...
 			target = null
 			path = list()
 		return
-	if(path.len)
+	else
 		step_to(src, path[1])
 		path -= path[1]
-		return 1
+		return TRUE
 	return
```
Comment (Geevies) [inline code/modules/mob/living/bot/cleanbot.dm:91]:
```suggestion
```

File: code/_onclick/click.dm [hunk 2]
```diff
@@ -70,6 +70,14 @@
 		var/mob/living/heavy_vehicle/M = loc
 		return M.ClickOn(A, params, src)
 
+	// pAI handling
+	if(istype(loc.loc, /mob/living/bot))
```
Comment (NonQueueingMatt) [inline code/_onclick/click.dm:74]: why loc.loc again?

Comment (Geevies) [reply code/_onclick/click.dm:74]: the pAI is inside a bot, inside a card. two locs.

**Pull request merged**